common/models/user.js

```diff
@@ -360,20 +367,19 @@ module.exports = function(User) {
     // Email model
     var Email = options.mailer || this.constructor.email || loopback.getModelByType(loopback.Email);
 
-    crypto.randomBytes(64, function(err, buf) {
-      if (err) {
-        fn(err);
-      } else {
-        user.verificationToken = buf.toString('hex');
-        user.save(function(err) {
-          if (err) {
-            fn(err);
-          } else {
-            sendEmail(user);
-          }
-        });
-      }
-    });
+    // Set a default token generation function if one is nto provided
+    var tokenGenerator = options.generateVerificationToken || User.generateVerificationToken;
+
+    tokenGenerator.apply(user, [function(token) {
```

Comment:
I really want to make the `generateVerificationToken` function consistent between the default one and custom function. If we think the `user` object is helpful for generating verification token (which I agree), I suggest that we change the signature to be `generateVerificationToken(user, cb)`.